scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala

```diff
@@ -28,14 +28,15 @@ case class Sketched[K, V](pipe: TypedPipe[(K, V)],
   seed: Int)(implicit serialization: K => Array[Byte],
     ordering: Ordering[K])
   extends HasReducers {
+  import CMSHasherImplicits._
 
   val reducers = Some(numReducers)
 
   private lazy val murmurHash = MurmurHash128(seed)
   def hash(key: K): Long = murmurHash(serialization(key))._1
 
-  private lazy implicit val cms = CMS.monoid(eps, delta, seed)
-  lazy val sketch: TypedPipe[CMS] =
+  private lazy implicit val cms = CMS.monoid[Long](eps, delta, seed)
```

Comment:
Those CMS values are not persisted beyond a single join.  Of course,
someone could do something crazy and save them, but the migration is their
problem at that point.

On Tue, Jan 27, 2015 at 11:06 AM, Alex Levenson <notifications@github.com>
wrote:

> In scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala
> <https://github.com/twitter/scalding/pull/1174#discussion_r23642791>:
>
> >
> >    val reducers = Some(numReducers)
> >
> >    private lazy val murmurHash = MurmurHash128(seed)
> >    def hash(key: K): Long = murmurHash(serialization(key))._1
> >
> > -  private lazy implicit val cms = CMS.monoid(eps, delta, seed)
> > -  lazy val sketch: TypedPipe[CMS] =
> > +  private lazy implicit val cms = CMS.monoid[Long](eps, delta, seed)
>
> Does this have any implications for persisted data, or is this OK to
> change?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/scalding/pull/1174/files#r23642791>.
>



-- 
Oscar Boykin :: @posco :: http://twitter.com/posco